docs/user-guides/client-api-config.md

```diff
@@ -0,0 +1,676 @@
+		Vector: &payload.Object_Vector{
+			// Set the vector
+			Vector: []float32{0, 1, 2},
+			// Set the specific ID for the vector
+			Id:     "sample",
+		},
+		// Upsert configuration (optional)
+		Config: &payload.Upsert_Config{
+			SkipStrictExistCheck: false,
+			Filters: &payload.Filter_Config{
+				Targets: []*payload.Filter_Target{
+					{
+						Host: "vald-ingress-filter",
+						Port: 8081,
+					},
+				},
+			},
+			Timestamp: time.Now().UnixMilli(),
+		},
+	})
+	if err != nil {
+		panic(err)
+	}
+        ...
+}
+```
+
+</details>
+
+#### skip_strict_exist_check
+
+`skip_strict_exist_check` (default value is `false`) is the flag for checking whether the same set of the vector and ID is already inserted or not.
+If it is set as `false`, the checking function will be skipped.<BR>
+
+When `skip_strict_exist_check` is `false`, the following checking steps will run in the upsert process:
+
+1. Whether the set of (ID and vector) is already inserted or not.
+   The request ID and vector will be inserted if there is no data.
+1. After passing the step.1, check whether the request vector is the same as the indexed vector.
+   If it is the same, the upsert process ends with returning the `ALREADY_EXIST` error.
+
+The upsert process will continue if all of the above steps have been passed.
+
+#### filters
+
+`filters` is the configuration when using filter functions.
+In the `Upsert` section, it is popular to use ingress filtering.
+
+The detailed configuration is following.
+
+```rpc
+// Filter related messages.
+message Filter {
+
+  // Represent the target filter server.
+  message Target {
+    // The target filter component hostname.
+    string host = 1;
+    // The target filter component port.
+    uint32 port = 2;
+  }
+
+  // Represent filter configuration.
+  message Config {
+    // Represent the filter target configuration.
+    repeated Target targets = 1;
+  }
+}
+```
+
+#### timestamp
+
+`timestamp` is the timestamp when the vector is inserted or updated.
+When `timestamp` is not set, the current time will be used.
+
+
+## Search Service
+
+Vald provides four types of search services.
+
+1. Search
+
+   - `Search` is the `ANN(Approximate Nearest Neighbor)` search with query vector.
+     It is a fast search even though large dimension vector.
+     The search duration is quick but less accurate than `LinearSearch`.
+     The search algorithm depends on each core algorithm.
+
+1. SearchById
+
+   - `SearchById` is the `ANN(Approximate Nearest Neighbor)` search with the stored vector's ID.
+     The ID should already exist in the NGT indexes before the search process.
+     The search algorithm is the same as `Search`.
+
+1. LinearSearch
+
+   - `LinearSearch` is the primary search algorithm with a query vector.
+     It searches all indexed vectors and calculates the distance between the query.
+     Its accuracy is exact, but the search time requires more than `Search` (ANN search) and increases the amount of indexed vector.
```

Comment:
What is `increases the amount of indexed vector` means here?

Reply:
```suggestion
     Its accuracy is exact, but the search time requires more than `Search` (ANN search) as increasing the amount of indexed vector.
```

Reply:
Sorry but I can't understand. what is increased when searching?

Reply:
```suggestion
     It returns accurate results but requires more processing time than `Search` (ANN search) as it needs to calculate from all indexed vectors instead of only a subset of indexed vectors.
```